### DISCUSSION TITLE
glibc 2.28 support?

### DISCUSSION THREAD

**Discussion by ganzgustav22** [rustdesk/rustdesk#1525] [General]
Do you really need to link against glibc 2.29? Is 2.28 from Debian buster not good somehow? Is there some feature present in glibc 2.29 that isn't there in 2.28 that you absolutely need to use because otherwise Rustdesk wouldn't work at all our would be insecure?

I would really like to use Rustdesk on a Raspberry Debian Buster system with glibc 2.28.

**Comment by rihani** [comment]
Did you get this working by any chance?

**Comment by ganzgustav22** [comment]
This "we always use the latest version, just because"-thing plus the fact that the question about future licensing (https://github.com/rustdesk/rustdesk/discussions/1440) stays unanswered has led me to believe that this is not a trustworthy project.

Even if it may somehow work on 2.28 (or if I'll maybe upgrade my system to 2.29 one day) I'm too afraid that they will either come up with some (unneeded) new version requirement in the future or they will start charging money, or cut down on features on the opensource version, etc.

Edit: Plus the fact that this software uses Sciter, a commercial software distributed in binary form where you never know what they will be up to in the future commercially and you have no idea what the binary code is actually doing (https://github.com/rustdesk/rustdesk/discussions/1528).

I have no trust in their technical decisions and no trust in their commercial decisions, won't use this software.

**Reply by Kingtous** [reply]
Finally, I successfully build the rustdesk shared library by compiling some dependencies by source, which can downgrade glibc dependency to `2.27`(ubuntu 18.04). We are struggling to loose this restriction because of the compilation issues from some of libraries. Once it's tested, we will release the arm/arm64 nightly build which only requires `glibc >= 2.27`.
